Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -46,28 +45,39 @@ protected static IMixedRealityInputSystem InputSystem
             }
         }
 
-        protected readonly List<IMixedRealityPointer> focusingPointers = new List<IMixedRealityPointer>();
-
         /// <summary>
         /// Pointers that are focusing the interactable
         /// </summary>
         public List<IMixedRealityPointer> FocusingPointers => focusingPointers;
+        protected readonly List<IMixedRealityPointer> focusingPointers = new List<IMixedRealityPointer>();
 
-        protected readonly HashSet<IMixedRealityInputSource> pressingInputSources = new HashSet<IMixedRealityInputSource>();
         /// <summary>
         /// Input sources that are pressing the interactable
         /// </summary>
         public HashSet<IMixedRealityInputSource> PressingInputSources => pressingInputSources;
-        
+        protected readonly HashSet<IMixedRealityInputSource> pressingInputSources = new HashSet<IMixedRealityInputSource>();
+
         /// <summary>
         /// Is the interactable enabled?
         /// </summary>
         public bool Enabled = true;
 
+        [FormerlySerializedAs("States")]
+        [SerializeField]
+        private States states;
```

Comment:
I've been testing this out, I was wondering, why does interactable have a States field, but then also have a separate States field for each theme? It sounds like it's possible for themes to have a state machine different from interactable. How is the States field in Interactable itself used? Is it used for event receivers?

![image](https://user-images.githubusercontent.com/25975362/65271312-c33bb180-dad1-11e9-92c0-f3803b4f0684.png)

Reply:
So this is part of the original architecture except both States properties were not surfaced.
Theme => Uses it's states property to determine how many values a property needs to define (for example, what color to set for each state)
Interactable => Uses States more to determine what kind of State Machine (i.e state model class) to instantiate at runtime inside it's class. Then it updates that state machine instance.

I throw a warning that the States properties need to be the same. Technically they don't because the values are accessed just based on index. So if I had a states object with just Default/Focus, it should theoretically work against an object with Default/Focus/Pressed/Disabled.

States in general requires a lot of refactoring. This is just something I didn't want to untangle and add risk/work to this PR.

It's weird but short term I don't know how to fix it. I can't just enforce only "DefaultInteractableStates" because we actually use the "FocusPressTouchGrabStates" as well as Interactable.

Reply:
The theme should hide the State field when rendered in the Interactable UI. It's part of Themes for the reason you mentioned above, to allow the theme to have something to configure without being placed on an Interactable.

From what I remember, there should be an issue with placing a theme with one States object on an Interactable with another States object containing different states. Or have a single theme applied to two Interactables with different state objects. Haven't tested that though.

Reply:
I changed it to not hide. I want it to be obvious to the user. Also, I am trying to veer away from custom render logic. It should just be here is a Theme and render like it you normally would. If the States objects are different, it renders a warning message. 

Furthermore, if I hide it, the user has to double click into the theme to change it anyways which is just more work.

Reply:
Good point. 

"If the States objects are different, it renders a warning message." this is cool.